lib/pseudolocalization.rb

```diff
@@ -31,6 +31,13 @@ def translate(locale, key, options)
         ::Pseudolocalization::I18n::Pseudolocalizer.pseudolocalize(original_backend.translate(locale, key, options))
       end
 
+      def translations
+        original = original_backend.translations
+        original.transform_values do |locale_translations|
+          pseudolocalize_node(locale_translations)
+        end
+      end
```

Comment:
Good point. Looking at the [ActiveRecord backend](https://github.com/svenfuchs/i18n-active_record/blob/master/lib/i18n/backend/active_record.rb#L75-L82), it does seem to take the approach of initializing the translations once in `init_translations` and simply returning them later for calls to `translations` (unless the `do_init` argument is passed).

Thoughts, @aergonaut? I suppose one reason to leave this as-is is that we don't necessarily know what the `original_backend` is doing in its `translations` method. The backend could be something other than the ActiveRecord one, so assuming it initializes and returns translations the same way as that code might not be good.

Reply:
> wouldn't this mean we'll transform the values every time we fetch a translation

As far as I can tell, no, this would not be called every time a translation is fetched. This library is a layer on top of the `original_backend`. When `translate` is called, it delegates to `original_backend.translate`[^1]. I believe delegating to the `original_backend` means that if the original were to call `translations`, it would be calling its own (unaltered) version.

[^1]: https://github.com/Shopify/pseudolocalization/blob/8fae663d1b8eb93512d47cdb9ac2a182a59c9e21/lib/pseudolocalization.rb#L29

> I suppose one reason to leave this as-is is that we don't necessarily know what the original_backend is doing in its translations method. The backend could be something other than the ActiveRecord one, so assuming it initializes and returns translations the same way as that code might not be good.

Simple's `init_translations` reads from the locale YAML files and stores them in a Hash. Doing the transform at that point and storing it into the underlying Hash could work.

However, KeyValue's is a no-op[^2]. So if we moved the transform to `init_translations`, if `original_backend` was a KeyValue, the translations would not actually be initialized at that point. KeyValue's implementation of `translations` also doesn't use a persistent Hash. It actually builds a new Hash every time it is called. So to work with KeyValue, we have to transform `translations` every time it is called.

[^2]: https://github.com/ruby-i18n/i18n/blob/55c7750a79aff26caa1a0b053c3a5b4432d23160/lib/i18n/backend/key_value.rb#L124-L130

I don't mind reworking this, if that's what's wanted!